### ISSUE TITLE
No exported method to free reader

### ISSUE THREAD

**Issue by shabbyrobe** [tinylib/msgp#179]
If you create a msgp Reader using `msgp.NewReader()`, it will acquire a reader from the `readerPool` but there's no way to return it - `freeR()` is unexported, as is the pool.

**Comment by peterbourgon** [comment]
This is a major bug! Can this get some attention?

**Comment by philhofer** [comment]
@peterbourgon I don't see how this is a "major bug;" it looks like a missing optimization at worst.
The pool doesn't retain a reference to the reader, so it is simply GC'd as normal, unless you've noticed something that I haven't.

**Comment by peterbourgon** [comment]
You're right, I mis-remembered how sync.Pool worked; I thought it retained references to leased objects. Forgive the alarm.
